### ISSUE TITLE
Codify bonded roles

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/proposals#53]
> _This is a Bisq Network proposal. Please familiarize yourself with the [submission and review process](https://docs.bisq.network/proposals.html)._

## Overview
We will maintain the bonded roles inside of the DAO. For applying to a bonded role the user need to make a request and if accepted by voting he need to lock up the required bond as defined by the role.

Only after the bond is locked up the role becomes active.

For many of the roles there will not be any code execution for validating that, as most roles are outside of the scope of the Bisq application. Exceptions are the arbitrators and mediators where the validation if one can be selected as arbitrator or mediator is derived from the DAO state.

We only cover here the roles where the role owner can cause serious damage to Bisq in case of abuse.

Here are the entries from the `bisq.core.dao.governance.role BondedRoleType` class.
First parameter is BSQ bond, second unlocktime in days, then the GH link to the role desciption and last if the role can be held by multiple persons.

```
     GITHUB_ADMIN(50_000, 60, "https://bisq.network/roles/16", true),
    FORUM_ADMIN(20_000, 60, "https://bisq.network/roles/19", true),
    TWITTER_ADMIN(20_000, 60, "https://bisq.network/roles/21", true),
    ROCKET_CHAT_ADMIN(20_000, 60, "https://bisq.network/roles/79", true),
    YOUTUBE_ADMIN(10_000, 60, "https://bisq.network/roles/56", true),

    // maintainers
    BISQ_MAINTAINER(50_000, 60, "https://bisq.network/roles/63", true),
    BITCOINJ_MAINTAINER(20_000, 60, "https://bisq.network/roles/8", true),
    NETLAYER_MAINTAINER(20_000, 60, "https://bisq.network/roles/81", true),

    // operators
    WEBSITE_OPERATOR(50_000, 60, "https://bisq.network/roles/12", true),
    FORUM_OPERATOR(50_000, 60, "https://bisq.network/roles/19", true),
    SEED_NODE_OPERATOR(20_000, 60, "https://bisq.network/roles/15", true),
    DATA_RELAY_NODE_OPERATOR(20_000, 60, "https://bisq.network/roles/14", true),
    BTC_NODE_OPERATOR(5_000, 60, "https://bisq.network/roles/67", true),
    MARKETS_OPERATOR(20_000, 60, "https://bisq.network/roles/9", true),
    BSQ_EXPLORER_OPERATOR(20_000, 60, "https://bisq.network/roles/11", true),
    MOBILE_NOTIFICATIONS_RELAY_OPERATOR(20_000, 60, "https://bisq.network/roles/82", true),

    // other
    DOMAIN_NAME_HOLDER(50_000, 60, "https://bisq.network/roles/77", false),
    DNS_ADMIN(20_000, 60, "https://bisq.network/roles/18", false),
    MEDIATOR(10_000, 60, "N/A", true),
    ARBITRATOR(200_000, 60, "https://bisq.network/roles/13", true),
    BTC_DONATION_ADDRESS_OWNER(20_000, 60, "https://bisq.network/roles/80", true);
```

**Comment by mrosseel** [comment]
I see no IRC role, is this an oversight or on purpose. No strong opinion either way.

**Comment by ManfredKarrer** [comment]
I intentionally only added the most critical roles where real damage can be done. Of course that is up to discussion where to draw that line - there is no hard line here... Just wanted to avoid to overload the DAO with things which are really required. I also did not add all repository maintainers. Not sure which path to go here...

Maybe @cbeams can add his opinion as well....

**Comment by cbeams** [comment]
> Maybe @cbeams can add his opinion as well....

The list and bond values look fine to me. I probably would have included `IRC_ADMIN`, but not critical.

How difficult will it be to add / remove / change / rename these roles and their values post-genesis?

**Comment by ManfredKarrer** [comment]
I still need to check out how flexible it will be with changes. We could use the Params voting for changing values but I don't want to over use that, otherwise we get 100s of parameters...
So I first want to find out how backward compatible changes will be and what are the "costs". I hope adding is not a big problem and then we could maybe  have updated versions like Mediator_v1, Mediator_v2.... But still requires more work to get there. I am atm in the finishing phase of the roles/bonds/confiscation domain. As soon that is complete i will check out update mechanisms....

**Comment by KanoczTomas** [comment]
Hm perhaps not doing a static coding of roles but having some kind of role ID and have a `lookup` table for it somewhere. The link between the ID and role could be somehow recorded on the blockchain, that would mean we could change it by vote, but would still be immutable. I have no idea how to do it, just thinking out laud. I am not sure having the role name and github link is the right way to do it. Sadly have no solution yet for it.

**Comment by ManfredKarrer** [comment]
@KanoczTomas Good idea! I will think about implementation feasibility....

**Comment by ManfredKarrer** [comment]
I am wondering if we should add roles for those as well:

- Mobile notification relay operator (ensure the operator is not logging)	
- Netlayer maintainer (we rely on the that library for Tor)	
- Bisq-BitcoinJ fork maintainer (we rely on the that library for the wallet)

Regarding the bond amounts in the above suggestions:
I made a rough overview table of who is currently the role owner and how much those contributors have roughly earned in BSQ so far and some of those roles would exceed the earned BSQ. The amounts are based on the 1 BSQ = 1 USD price. If BSQ price is 5 or 10 USD the bond are way too high. So probably best is to wait a few months before we start locking up bonds for roles and then use adjusted bond amounts.

I will check if and how we can adjust the bond amount without breaking consensus.

**Comment by ManfredKarrer** [comment]
Bond amounts can be changed without breaking anything.
See https://github.com/bisq-network/bisq/issues/2559

**Comment by ManfredKarrer** [comment]
Added 
BTC_DONATION_ADDRESS_OWNER(20_000, 60, "https://bisq.network/roles/80", true);

**Comment by devinbileck** [comment]
To me, the BSQ amount for FORUM_ADMIN and YOUTUBE_ADMIN seem low (based on 1 BSQ = 1 USD).

**Comment by ManfredKarrer** [comment]
Ups FORUM_ADMIN was wrong, is 20 000. Just corrected it. Yes YOUTUBE_ADMIN might be too low.

**Comment by ManfredKarrer** [comment]
I updated above list and added 3 new roles and adjusted bond for DNS.

**Comment by ManfredKarrer** [comment]
Increase YOUTUBE_ADMIN to 10 000 as well

**Comment by ManfredKarrer** [comment]
I would say lets close that if noboday has further concerns.

**Comment by ManfredKarrer** [comment]
I will update BTC_DONATION_ADDRESS_OWNER to 50 000 BSQ to reflect better the potential amount there.

**Comment by ManfredKarrer** [comment]
I know that is closed byt just for documentation reasons: 

After a discussion with @eyalron33 we came to the conclusion that 75 days lockup time for bonds might be too short. In case of a sabotage attack where a voting round is rendered invalid a confiscation request could be delayed until the locked up bonds are unlocked. To avoid that I think we should increase it to 110 days, that is about 3,5 months and should give more protection. 
It also adds more protection value to the bond as a longer lock time represents that the role owner is more trustworthy and long term invested.

**Comment by sqrrm** [comment]
I'm fine with 110 days, but please make a new PR to handle the merge and just add a reference to this PR there.

**Comment by ManfredKarrer** [comment]
Its in the rc1.0 branch. I did not want to risk anything with current dao_regtest.
